project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/settings/AppSettingsComponent.kt

```diff
@@ -207,12 +236,59 @@ class AppSettingsComponent {
             .addSeparator()
             .addLabeledComponent(JBLabel("Transformations:"), createTransformationPanel(), 1, false)
             .addSeparator()
+            .addComponent(JBLabel("Add here directories/files of your project if you are sure that they are important or can not be minimized").apply {
+                border = BorderFactory.createEmptyBorder(10, 0, 10, 0)
+            })
+            .addComponent(pathsPanel)
+            .addSeparator()
             .addComponentFillVertically(JPanel(), 0)
             .panel
 
         updateUIState()
     }
 
+    private fun createPathPanel(): JPanel {
+        pathTable.columnModel.getColumn(0).apply {
+            preferredWidth = 700  // Width of first column
+        }
+
+        val toolbarDecoratorPanel = ToolbarDecorator.createDecorator(pathTable)
+            .disableUpAction()
+            .disableDownAction()
+            .setAddAction {
+                val chosenFiles = FileChooser.chooseFiles(fileChooserDescriptor, null, null)
+                for (file in chosenFiles) {
+                    val path = file.path
+                    if ((0 until pathTableModel.rowCount).none { pathTableModel.getValueAt(it, 0) == path }) {
+                        pathTableModel.addRow(arrayOf(path))
+                    }
+                }
+            }
+            .setRemoveAction {
+                val selectedRows = pathTable.selectedRows.sortedDescending()
+                for (row in selectedRows) {
+                    pathTableModel.removeRow(row)
+                }
+            }
+            .createPanel()
+
+        return JPanel(BorderLayout()).apply {
+            add(toolbarDecoratorPanel, BorderLayout.CENTER)
+        }
+    }
+
+    private fun toRelativePath(absolutePath: String): String = if (projectBaseDir.isNotBlank() && absolutePath.startsWith(projectBaseDir)) {
+        absolutePath.removePrefix("$projectBaseDir/")  // Убираем корневую директорию
+    } else {
+        absolutePath
+    }
+
+    private fun toAbsolutePath(relativePath: String): String = if (projectBaseDir.isNotBlank()) {
+        "$projectBaseDir/$relativePath"
+    } else {
+        relativePath
+    }
```

Comment:
I suppose that after some fixes, we don't need these functions.
But if they are needed, please consider to use standard nio path functions to perform such actions. It will allow us to consider not only unix-like systems